service/resource_router.go

```diff
@@ -0,0 +1,33 @@
+package service
+
+import (
+	"context"
+
+	"github.com/giantswarm/kvm-operator/service/keyv2"
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/operatorkit/framework"
+)
+
+// newResourceRouter determines which resources are enabled based upon the
+// version in the version bundle.
+func newResourceRouter(resources []framework.Resource) func(ctx context.Context, obj interface{}) ([]framework.Resource, error) {
+	return func(ctx context.Context, obj interface{}) ([]framework.Resource, error) {
+		var enabledResources []framework.Resource
+
+		customObject, err := keyv2.ToCustomObject(obj)
+		if err != nil {
+			return enabledResources, microerror.Mask(err)
+		}
+
+		switch keyv2.VersionBundleVersion(customObject) {
+		case "0.1.0":
+			enabledResources = resources
+		case "1.0.0":
+			enabledResources = resources
+		default:
+			return enabledResources, microerror.Maskf(invalidVersionError, "version '%s' in version bundle is invalid", keyv2.VersionBundleVersion(customObject))
```

Comment:
Nitpick, the problem with the version is rather that it is not support or does not exist as that it is invalid.

Reply:
What will happen then if the version does not exist? Do we return default resources or nil? Because I am not sure how the framework will work with it.

Reply:
The concept of version bundles implies some guarantee. An operator exposes version bundles and guarantees some business logic. When something is requested that is not guaranteed an error should be returned. So we return an error still as you do already.